plugin/builtin/configmapgenerator/ConfigMapGenerator.go

```diff
@@ -8,32 +8,72 @@ import (
 	"sigs.k8s.io/kustomize/api/kv"
 	"sigs.k8s.io/kustomize/api/resmap"
 	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/errors"
 	"sigs.k8s.io/yaml"
 )
 
 type plugin struct {
-	h                *resmap.PluginHelpers
+	h *resmap.PluginHelpers
+
 	types.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 	types.ConfigMapArgs
 }
 
 //noinspection GoUnusedGlobalVariable
 var KustomizePlugin plugin
 
-func (p *plugin) Config(h *resmap.PluginHelpers, config []byte) (err error) {
+func (p *plugin) Config(h *resmap.PluginHelpers, config []byte) error {
 	p.ConfigMapArgs = types.ConfigMapArgs{}
-	err = yaml.Unmarshal(config, p)
-	if p.ConfigMapArgs.Name == "" {
-		p.ConfigMapArgs.Name = p.Name
+	p.h = h //TODO: deep copy
```

Comment:
These deep copy TODOs are waiting on #4654, though some can technically be avoided by storing parts of the `PluginHelpers`.